### PR TITLE
fix: scroll to last message

### PR DESCRIPTION
**Pull request by anfibiacreativa**
## Purpose

Scroll to bottom of chat list when the messages update

- ...

## Does this introduce a breaking change?

```
[ ] Yes
[ x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

- Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

- Test the code

Submit a question. Chat auto scrolls smoothly and the chat form doesn't overlap

## What to Check

Performance is not degraded since event is debounced.
perf
perf1

### CODE REVIEW

File: packages/chat-component/src/main.ts [hunk 1]
```diff
@@ -105,8 +105,30 @@ export class ChatComponent extends LitElement {
...
+  debounceDispatchMustScrollEvent(): void {
+    let timeout: any = 0;
+    clearTimeout(timeout);
+    timeout = setTimeout(() => {
+      this.dispatchEvent(mustScrollEvent);
+    }, 500);
+  }
+  // handle must scroll event
+  handleOnMustScroll(): void {
+    const footer = this.shadowRoot?.querySelector('#chat-list-footer') as HTMLElement;
```
Comment (manekinekko) [inline packages/chat-component/src/main.ts:126]: is there an idiomatic `lit`-way to do this?

Comment (manekinekko) [reply packages/chat-component/src/main.ts:126]: Looks we can use `@query()` https://lit.dev/docs/components/shadow-dom/#query

Comment (anfibiacreativa) [reply packages/chat-component/src/main.ts]: Fixed!

File: packages/chat-component/src/main.ts [hunk 2]
```diff
@@ -105,8 +108,29 @@ export class ChatComponent extends LitElement {
 
   static override styles = [mainStyle];
 
-  // Send the question to the Open AI API and render the answer in the chat
+  override connectedCallback(): void {
+    super.connectedCallback();
 
+    // add event listener to know when to scroll chat list footer into view
+    this.addEventListener('must-scroll', () => {
```
Comment (sinedied) [inline packages/chat-component/src/main.ts:115]: Why use an event for that, and not perform the scroll directly in the debounced call?

Is this something that we want to expose to the outside?

Comment (anfibiacreativa) [reply packages/chat-component/src/main.ts:115]: I didn't initially implement it as an event, however considering that the application is meant to be integrated as part of a shell application, I considered posting the event as a message to do things like closing menus or other type of UI improvements in the shell. We can handle that as a different issue and trigger and only once event when the item is into view already. Removed all the utils and event.

**Review by sinedied — APPROVED** [review]

**Pull request merged**